tgui/packages/tgui/interfaces/MessageMonitor.tsx

```diff
@@ -0,0 +1,380 @@
+import { BooleanLike } from 'common/react';
+import { useBackend, useLocalState } from '../backend';
+import { Section, Stack, Input, Button, Table, NoticeBox, Box } from '../components';
+import { Window } from '../layouts';
+
+enum Screen {
+  Main,
+  MessageLogs,
+  RequestLogs,
+  Hacked,
+}
+
+type Data = {
+  screen: Screen;
+  status: BooleanLike;
+  server_status: number;
+  auth: number;
+  password: string;
+  is_malf: BooleanLike;
+  error_message: string;
+  success_message: string;
+  notice_message: string;
+  requests: Request[];
+  messages: Message[];
+};
+
+type Request = {
+  ref: string;
+  message: string;
+  stamp: string;
+  send_dpt: string;
+  id_auth: string;
+};
+
+type Message = {
+  ref: string;
+  message: string;
+  sender: string;
+  recipient: string;
+};
+
+const RequestLogsScreen = (props, context) => {
+  const { act, data } = useBackend<Data>(context);
+  const { requests } = data;
```

Comment:
```
  const { requests = [] } = data;
```
I prefer casting in the destructure statement to using an optional chain later, there's no set rule for this though